Experimental/protecting libE fields

jmlarson1 opened the pull request:
This proposed change checks that any field returned from a `sim_f` or `gen_f` is not protected. (We could do a check in `check_inputs` that `gen_specs['out']` and `sim_specs['out']` aren't protected.)

Also, the default is to deepcopy the protected libE fields and ensure they aren't changed by the `alloc_f`. (This can be overwritten by setting `libE_specs['safe_mode'] = False`)

shuds13 reviewed (approved):
Presumably last line should sat `libE_specs['safe_mode'] = False`

When `cancel_requested` is added we can make an exception.

Is there a simple way to unit test this?

jmlarson1 commented:
> Is there a simple way to unit test this?

I've added a unit test and removed `sim_id` from the protected field names as that can be given by example `gen_f`
